Update Release Schedule Section Pertaining to Patch Releases (#1387)
nautobot/nautobot: nautobot/docs/development/index.md

  - The final notification will be provided 3-5 days before the release drops.
- The dates and notifications will occur by updating the GitHub Release Milestone and on Slack.

#### Patch Releases

Specifics around patch releases:

- Will be released every two weeks on Mondays, starting March 7th, 2022.
- Will be focused on bugs, security vulnerabilities, backports, and other issues as they arise.

> - Will not introduce new functionality that isn't addressing a bug.

Reviewer: ```suggestion
- Will not introduce significant new functionality except as needed to address a bug.
```

> - Will not be released if there has been no new merges to `develop` since the previous release.

Reviewer: ```suggestion
- Will not be released if there have been no new merges to `develop` since the previous release.
```

- Should not be considered a mandatory upgrade: If it does not fix a bug or issue you are experiencing, you should not feel the need to upgrade right away.

> - May happen a day or two after the Monday schedule if Monday is an observed holiday. We would like to ensure full availability post-release.

Reviewer: ```suggestion
- May happen a day or two after the Monday schedule if Monday is an observed holiday. We would like to ensure full team availability post-release.
```

> Should a patch release contain a fix for CVE(s), data-loss bug(s), or other vulnerability, we will:

Reviewer: ```suggestion
Should a patch release contain a fix for security vulnerability(s) (i.e. CVE(s)), data-loss bug(s), or other critical issue(s), we will:
```

- Release a new patch release as soon as the fix has been identified, implemented, and thoroughly tested. No waiting for the next regularly scheduled release date.
- Bring special attention to these releases in our notification channels.
- Strongly urge you to upgrade to address these more serious issues as soon as possible.

> - Not adjust any preceding/proceeding release dates.

Reviewer: ```suggestion
- Not adjust any subsequent release dates (the next scheduled release will still occur as scheduled).
```
